src/ray/rpc/gcs_server/gcs_rpc_client.h

```diff
@@ -225,14 +229,42 @@ class GcsRpcClient {
   }
 
   void Reconnect() {
+    absl::MutexLock lock(&mutex_);
     if (get_server_address_) {
-      auto address = get_server_address_();
-      Init(address.first, address.second, client_call_manager_);
+      std::pair<std::string, int> address;
+      int index = 0;
+      for (; index < RayConfig::instance().ping_gcs_rpc_server_max_retries(); ++index) {
+        address = get_server_address_();
+        if (Ping(address.first, address.second, 100)) {
+          RAY_LOG(INFO) << "Target rpc server " << address.first << ":" << address.second
+                        << " is valid.";
+          break;
+        }
+        usleep(RayConfig::instance().ping_gcs_rpc_server_interval_milliseconds() * 1000);
+      }
+
+      if (index < RayConfig::instance().ping_gcs_rpc_server_max_retries()) {
+        Init(address.first, address.second, client_call_manager_);
+        if (reconnected_callback_) {
+          reconnected_callback_();
+        }
+      } else {
+        RAY_LOG(FATAL) << "Target rpc server " << address.first << ":" << address.second
+                       << " is invalid.";
+      }
```

Comment:
"Couldn't reconnect to GCS server. The last attempted GCS server address was xxx."

Reply:
fixed